includes/class-wc-cart.php

```diff
@@ -1738,9 +1738,23 @@ public function add_discount( $coupon_code ) {
 
 		// If its individual use then remove other coupons
 		if ( $the_coupon->get_individual_use() ) {
-			$this->applied_coupons = apply_filters( 'woocommerce_apply_individual_use_coupon', array(), $the_coupon, $this->applied_coupons );
+			$coupons_to_keep = apply_filters( 'woocommerce_apply_individual_use_coupon', array(), $the_coupon, $this->applied_coupons );
+
+			foreach ( $this->applied_coupons as $applied_coupon ) {
+				$keep_key = array_search( $applied_coupon, $coupons_to_keep );
```

Comment:
You get the array key. :) 

Here's the algorithm for this chunk of code:
1. Get all of the coupons you want to keep.
2. If an existing coupon isn't a coupon you want to keep, remove it.
3. If an existing coupon is a coupon you want to keep, unset the entry in `$coupons_to_keep` (array key needed here)
4. Anything left in `$coupons_to_keep` is a new coupon you want added to the coupons.